deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/FilterableRepositoryImplementor.java

```diff
@@ -79,6 +116,115 @@ public ClassVisitor apply(String className, ClassVisitor classVisitor) {
                 listCreator.getMethodParam(1),
                 listCreator.loadClassFromTCCL(entityType.toString())));
 
+        implementIsNew(index, entityType, transformer);
+
         return transformer.applyTo(classVisitor);
     }
+
+    private void implementIsNew(IndexView index, DotName entityType, ClassTransformer transformer) {
+        ClassInfo filterableRepository = index.getClassByName(filterableRepositoryInterface);
+        MethodInfo isNewMethod = filterableRepository.method("isNew", TypeVariable.create("T"));
+        if (isNewMethod == null) {
+            return;
+        }
+
+        MethodCreator isNewCreator = transformer.addMethod(
+                isNewMethod.name(),
+                isNewMethod.returnType().toString(),
+                isNewMethod.parameterType(0).name().toString());
+        ResultHandle entity = isNewCreator.getMethodParam(0);
+        AnnotationTarget idAnnotationTarget = getIdAnnotationTarget(entityType, index);
+        Type idType = getTypeOfTarget(idAnnotationTarget);
+        ResultHandle idValue = generateObtainValue(isNewCreator, entityType, entity, idAnnotationTarget);
+        if (idType instanceof PrimitiveType) {
+            if (!idType.name().equals(PRIMITIVE_LONG) && !idType.name().equals(PRIMITIVE_INTEGER)) {
+                throw new IllegalArgumentException("Id type of '" + entityType + "' is invalid.");
+            }
+            if (idType.name().equals(PRIMITIVE_LONG)) {
+                ResultHandle longObject = isNewCreator.invokeStaticMethod(
+                        MethodDescriptor.ofMethod(Long.class, "valueOf", Long.class, long.class), idValue);
+                idValue = isNewCreator.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(Long.class, "intValue", int.class), longObject);
+            }
```

Comment:
Could be reduced to:
```suggestion
            if (idType.name().equals(PRIMITIVE_LONG)) {
                ResultHandle longObject = isNewCreator.invokeStaticMethod(
                        MethodDescriptor.ofMethod(Long.class, "valueOf", Long.class, long.class), idValue);
                idValue = isNewCreator.invokeVirtualMethod(
                        MethodDescriptor.ofMethod(Long.class, "intValue", int.class), longObject);
            } else if (!idType.name().equals(PRIMITIVE_INTEGER)) {
                throw new IllegalArgumentException("Id type of '" + entityType + "' is invalid.");
            } 
```
don't know which is easier to understand.

Reply:
I'm also not sure what is more readable.
The code is inspired by a similar method in Quarkus.
I propose to let it as is, at least for the moment